backend/src/main/java/com/google/blackswan/mock/SimpleAnomalyGenerator.java

```diff
@@ -0,0 +1,143 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.blackswan.mock;
+
+import static java.util.stream.Collectors.toMap;
+
+import com.google.models.*;
+import java.util.Scanner;
+import java.util.*;
+import java.text.ParseException;
+import java.util.logging.Logger;
+import java.io.IOException;
+import java.time.format.DateTimeParseException;
+import java.lang.Math;
+import java.lang.Object;
+
+/** Generate list of anomalies based data in the csv file using average and threshold. */
+public class SimpleAnomalyGenerator implements AnomalyGenerator {
+  private static final String DATA_FILE_LOCATION = "/sample-ramen-data.csv";
+  private static final String EXCEPTION_MESSAGE = "Invalid row format.";
+  private static final Logger log = Logger.getLogger(SimpleAnomalyGenerator.class.getName());
+  private static final int THRESHOLD = 13;
+  private static final int NUM_POINTS = 5;
+
+  private List<Anomaly> anomalies;
+  private Map<Timestamp, Integer> data;
+
+  public SimpleAnomalyGenerator() {
+    anomalies = new ArrayList<Anomaly>();
+    data = new LinkedHashMap<>();
+    parseCSV();
+    generateAnomalies();
+  }
+
+  private void generateAnomalies() {
+    int sum = 0;
+    for (int val : data.values()) {
+      sum += val;
+    }
+    int ave = sum / data.size();
+
+    // Find instances where exceed threshold.
+    Map<Timestamp, Integer> anomalyPoints = data.entrySet().stream()
+        .filter(e -> e.getValue() - ave > THRESHOLD)
+        .collect(toMap(e -> e.getKey(), e -> e.getValue()));
+
+    // Create anomaly objects from those instances.
+    for (Map.Entry<Timestamp, Integer> entry : anomalyPoints.entrySet()) {
+      anomalies.add(createAnomalyFromDataPoint(entry.getKey()));
+    }
+  }
+
+  /** TODO: Depending on size of future data, alter algorithm of finding associated data points. */
```

Comment:
optional: maybe be more specific about how you would alter the algorithm?

Reply:
I haven't thought of a better algorithm for super large data yet, will update once I have better idea :')